src/Native/Runtime/amd64/PInvoke.asm

```diff
@@ -341,4 +340,59 @@ LEAF_ENTRY RhpReversePInvokeReturn, _TEXT
 LEAF_END RhpReversePInvokeReturn, _TEXT
 
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;
+;; RhpPInvoke
+;;
+;; IN:  RCX: address of pinvoke frame
+;;
+;; TRASHES: R10, R11
+;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+LEAF_ENTRY RhpPInvoke, _TEXT
+        ;; R10 = GetThread(), TRASHES R11
+        INLINE_GETTHREAD r10, r11
+
+        ;; The callsite of the helper is as good to start the stackwalk as the actual PInvoke callsite
```

Comment:
This is not true unless the JIT has some magic to emit the partially interruptible root string at this address instead of the actual call site.  If there is magic, please make it clear in the function header how it's supposed to work and what pieces have what responsibilities.